centreon/www/front_src/src/Header/Poller/usePollerDatas.ts

```diff
@@ -0,0 +1,68 @@
+import { useState, useEffect, useMemo } from 'react';
+
+import { useNavigate } from 'react-router-dom';
+import { useAtomValue } from 'jotai/utils';
+import { useTranslation } from 'react-i18next';
+import { equals, isNil } from 'ramda';
+
+import { useFetchQuery } from '@centreon/ui';
+import { refreshIntervalAtom, userAtom } from '@centreon/ui-context';
+
+import useNavigation from '../../Navigation/useNavigation';
+import { pollerListIssuesEndPoint } from '../api/endpoints';
+import { pollerIssuesDecoder } from '../api/decoders';
+
+import { getPollerPropsAdapter } from './getPollerPropsAdapter';
+import type { GetPollerPropsAdapterResult } from './getPollerPropsAdapter';
+
+interface UsePollerDatasResust {
+  data: GetPollerPropsAdapterResult;
+  error: unknown;
+  isAllowed: boolean;
+  isLoading: boolean;
+}
+
+export const usePollerDatas = (): UsePollerDatasResust => {
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+  const { allowedPages } = useNavigation();
+  const { isExportButtonEnabled } = useAtomValue(userAtom);
+  const refetchInterval = useAtomValue(refreshIntervalAtom);
+  const [datas, setDatas] = useState(null);
+  const [isAllowed, setIsAllowed] = useState<boolean>(true);
+
+  const { isLoading, data } = useFetchQuery({
+    catchError: ({ statusCode }): void => {
+      if (equals(statusCode, 401)) {
+        setIsAllowed(false);
+      }
+    },
+    decoder: pollerIssuesDecoder,
+    getEndpoint: () => pollerListIssuesEndPoint,
+    getQueryKey: () => [pollerListIssuesEndPoint, 'get-poller-status'],
+    queryOptions: {
+      refetchInterval: refetchInterval * 1000
+    }
+  });
+
+  useEffect(() => {
+    if (!isNil(data)) {
+      setDatas(
+        getPollerPropsAdapter({
+          allowedPages,
+          data,
+          isExportButtonEnabled,
+          navigate,
+          t
+        })
+      );
+    }
+  }, [data]);
```

Comment:
I think you can get rid of the state for the data because, you just have to return the adapted data. Gonna spare a committed render

Reply:
is there an option within `useFetchQuery` that would allow me that ? 
because I need to test `!isNil(data)` before returning, I don't feel the return is the right spot to do it, also because the `useMemo` prevent any subscriber re-rendering in theory.

Reply:
data will be undefined or populated with data from the API. Either you return undefined (means you haven't done your API request yet or you return your adapted data

Reply:
check, is that what you had in mind ?